yarn-project/Earthfile

```diff
@@ -1,97 +1,88 @@
-        yarn workspaces focus @aztec/cli @aztec/aztec --production && \
-        cp -r node_modules node_modules_cache
-    SAVE ARTIFACT /build
+    RUN ./bootstrap.sh full
 
-# TODO versioning flow at end before publish?
-# ENV COMMIT_TAG=$EARTHLY_BUILD_SHA
-# RUN ./scripts/version_packages.sh
+build-prod:
+    FROM +build
+    RUN yarn workspaces focus @aztec/cli @aztec/aztec --production && yarn cache clean
+    SAVE ARTIFACT /usr/src /usr/src
 
-# run:
-#     FROM node:18.19.1-slim
-#     ARG COMMIT_TAG=""
-#     ENV COMMIT_TAG=$COMMIT_TAG
-#     COPY --from=builder /build /build
-#     WORKDIR /build/yarn-project
-#     ENTRYPOINT ["yarn"]
+build-prod-slim:
+    FROM node:18.19.1-slim
+    COPY +build-prod/usr/src /usr/src
 
 aztec:
-    FROM +build
-    # ENV vars for using native ACVM simulation
+    FROM +build-prod-slim
     ENV ACVM_BINARY_PATH="/usr/bin/acvm" ACVM_WORKING_DIRECTORY="/tmp/acvm"
-    ENTRYPOINT ["node", "--no-warnings", "/build/yarn-project/aztec/dest/bin/index.js"]
+    ENTRYPOINT ["node", "--no-warnings", "/usr/src/yarn-project/aztec/dest/bin/index.js"]
     EXPOSE 8080
-    # TODO(AD) the following are the biggest node modules bundled, should they be deleted as they are build tools?
-    # 25840   @jest
-    # 31544   typescript
-    # 62368   @types
 
-source:
-    # for debugging rebuilds
-    RUN echo CONTENT HASH $(find . -type f -exec sha256sum {} ';' | sort | sha256sum | awk '{print $1}') | tee .content-hash
+cli:
+    FROM +build-prod-slim
+    ENTRYPOINT ["node", "--no-warnings", "/usr/src/yarn-project/cli/dest/bin/index.js"]
+    EXPOSE 8080
 
-end-to-end:
-    FROM +source
-    # compilation artifacts for end-to-end
-    # TODO encapsulate in bash script for cleanliness
-    RUN ./bootstrap.sh && \
-        yarn workspace @aztec/end-to-end run build:web && \
-        yarn workspaces focus @aztec/end-to-end --production \
-        && yarn cache clean
-    # Build web bundle for browser tests
-    RUN yarn workspace @aztec/end-to-end run build:web
+end-to-end-prod:
+    FROM +build
     RUN yarn workspaces focus @aztec/end-to-end --production && yarn cache clean
-    SAVE ARTIFACT /build
+    SAVE ARTIFACT /usr/src /usr/src
 
-end-to-end-minimal: 
-    # end to end test runner
-    BUILD +aztec
+end-to-end:
     FROM node:18.19.1-slim
     RUN apt-get update && apt-get install jq chromium -y
     ENV CHROME_BIN="/usr/bin/chromium"
-    COPY +end-to-end/build /build
-    WORKDIR /build/yarn-project/end-to-end
+    COPY +end-to-end/usr/src /usr/src
+    WORKDIR /usr/src/yarn-project/end-to-end
     ENTRYPOINT ["yarn", "test"]
 
-build-end-to-end:
-    ARG EARTHLY_GIT_HASH
-    FROM +end-to-end-minimal
-    SAVE IMAGE --push aztecprotocol/end-to-end-cache:$EARTHLY_GIT_HASH
-    FROM +aztec
-    SAVE IMAGE --push aztecprotocol/aztec-cache:$EARTHLY_GIT_HASH
\ No newline at end of file
+all:
+    BUILD +aztec
+    BUILD +cli
+    BUILD +end-to-end-slim
+
+# build-end-to-end:
+#     ARG EARTHLY_GIT_HASH
```

Comment:
build-end-to-end is used by the current CI framework
build-x86 ends in this step